packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -25,7 +26,8 @@ export class WindowsImage implements IMachineImageSource  {
    * Return the image to use in the given context
    */
   public getImage(scope: Construct): MachineImage {
-    const ami = Context.getSsmParameter(scope, this.imageParameterName(this.version));
+    const parameterName = this.imageParameterName(this.version);
```

Comment:
>The use of the SSM context provider by the EC2 and ECS libraries to retrieve AMIs was
replaced by deploy-time resolution of SSM parameters, so no fallback is required.

I think this is not great to do for all cases. The whole point of context was to do controlled updates. For example, if I just update my application to add a queue and deploy, or I change something to do a hotfix, is it okay that I pull in a new AMI with  a new version of Amazon Linux and a different version of libfoobar.so.5 which happens to  contain a bug and now my application is broken?

The point of context was that you would explicitly opt in to updates to "critical" parameters like that.

It's fine to fallback to the generic behavior (imo) only if we can't do anything else, so if the stack is region-agnostic. If the stack is region-aware, we can and should do better.

Reply:
https://github.com/awslabs/aws-cdk/issues/2933